hw_4_numpy/numpy_challenge.py

```diff
@@ -0,0 +1,48 @@
+import numpy as np
+
+if __name__ == "__main__":
+    first_array = np.zeros(10) 
+    second_array = np.array([['Argentina', 'Jamaica'], [5, 0]]) 
+    third_arrray = np.arange(18) 
+
+    #print(first_array, second_array, third_arrray, sep = '\n')
+
+
+
+def matrix_multiplication(matrix1, matrix2):
+    return np.matmul(matrix1, matrix2)
+
+def multiplication_check(matrix_lst): 
+    prev_matrix = matrix_lst[0]
+    can_multiply = True
+    for m in matrix_lst[1:]:
+        '''
+        multiplication check would fail for single matrices 
+        and if array is not a matrix (just an array)
+        also it would leave loop if can_multyply is already False
+        '''
+        if m.ndim != 2  or prev_matrix.ndim != 2 or len(matrix_lst) < 2 or can_multiply == False:
+            can_multiply = False
+            break
+        can_multiply = can_multiply and (prev_matrix.shape[1] == m.shape[0])
+        prev_matrix = m
```

Comment:
Сложное... Почему бы просто не
```suggestion
    for i in range(len(matrix_lst)-1):
        if matrix_lst[i].shape[1] != matrix_lst[i+1].shape[0]:
            return False
    return True
```

Reply:
я не совсем согласен, он же может выдать ошибку индексации, если использовать такой подход
плюс, если пользователь введет не двухмерный эррэй, а одномерный, например?  тогда shape[1] просто не будет и все выпадет в ошибку (у меня выпадало на моих тестовых данных, если вводить одномерный эррэй)

Reply:
@wwoskie, вообще подразумевалось, что функция должна работать с двумерными эрреями (по-моему даже в чате про это писали, но это не точно), но, в целом, да, я согласен, что нужно учитывать и одномерные. Сорян за буллинг)

Reply:
нет, спасибо за жизненный урок, теперь буду читать чат)